### PR TITLE
[specs] Clarification to derivation pipeline specs

### PR DESCRIPTION
**Pull request by norswap**
Built on top of https://github.com/ethereum-optimism/optimism/pull/3203

### CODE REVIEW

File: specs/derivation.md [hunk 1]
```diff
@@ -478,39 +486,56 @@ Some frames are ignored:
...
-> cf. slack discussion with Proto
+- The channel is pruned out of the channel bank (see below), in which case it isn't passed to the further stages.
+
+> **TODO** specify CHANNEL_TIMEOUT (currently 120s on Goerli testnet)
+
+As currently implemented, each step in this stage performs the following actions:
+
+- Try to prune the channel bank.
+    - This occurs if the size of the channel bank exceeds `MAX_CHANNEL_BANK_SIZE` (currently set to 100,000,000 bytes).
+    - The size of channel bank is the sum of the sizes (in btes) of all the frames contained within i.
```
Comment (trianglesphere) [inline specs/derivation.md:518]:
```suggestion
    - The size of channel bank is the sum of the sizes (in btes) of all the frames contained within it.
```

File: specs/derivation.md [hunk 2]
```diff
@@ -587,7 +612,9 @@ In particular, the following fields of the payload attributes are checked for eq
 If consolidation fails, the unsafe L2 head is reset to the safe L2 head.
 
 If the safe and unsafe L2 heads are identical (whether because of failed consolidation or not), we send the block to the
-execution engine to be converted into a proper L2 block, which becomes both the new L2 safe and unsafe heads.
+execution engine to be converted into a proper L2 block, which will become the new L2 unsafe head.
+
+TODO TODO TODO: above said "will become both new L2 safe and unsafe heads" -- does the code do this?
```
Comment (trianglesphere) [inline specs/derivation.md:617]: yes. when sending a payload building on top of the unsafe/safe head (if they are the same), we advance both. We never let the unsafe head get behind the safe head.

**Review by trianglesphere — COMMENTED** [review]
small comments, but generally looking good.

**Review by trianglesphere — APPROVED** [review]

**Review by mslipper — APPROVED** [review]

**Comment by norswap** [comment]
Owned by auto-merge, this was supposed to be changed to merge into develop after #3203 was merged.
Instead, this got merged into #3203 first.

I think the current setup doesn't really support staggered changes like this very well — you can't ensure that PRS are merged into `develop` in order, nor that they "fold" into each other in order (e.g. v3 merged in v2, v2 merged in v1, v1 merged in develop).

Anyway, I'll fold this into #3223.

**Pull request merged**